tests/test_openclhpp.cpp

```diff
@@ -3035,5 +3068,99 @@ void testDevicePCIBusInfo_KHR()
+static cl_int clGetCommandBufferInfoKHR_testCommandBufferKhrGetNumQueues(
+    cl_command_buffer_khr command_buffer,
+    cl_command_buffer_info_khr param_name,
+    size_t param_value_size,
+    void *param_value,
+    size_t *param_value_size_ret,
+    int num_calls)
+{
+    (void) num_calls;
+    TEST_ASSERT_EQUAL_PTR(make_command_buffer_khr(0), command_buffer);
+    TEST_ASSERT_EQUAL_HEX(CL_COMMAND_BUFFER_NUM_QUEUES_KHR, param_name);
+    TEST_ASSERT(param_value == NULL || param_value_size >= sizeof(cl_uint));
+    if (param_value_size_ret != NULL)
+        *param_value_size_ret = sizeof(cl_uint);
+    if (param_value != NULL)
+        *(cl_uint *) param_value = 1;
+    return CL_SUCCESS;
+}
+
+void testCommandBufferInfoKHRNumQueues()
+{
+#if defined(cl_khr_command_buffer)
+    cl_uint expected = 1;
+    int refcount = 1;
+    cl_int error;
+
+    clGetCommandBufferInfoKHR_StubWithCallback(clGetCommandBufferInfoKHR_testCommandBufferKhrGetNumQueues);
+
+    cl_uint num = commandBufferKhrPool[0].getInfo<CL_COMMAND_BUFFER_NUM_QUEUES_KHR>();
+    TEST_ASSERT_EQUAL_HEX(expected, num);
+#endif
+}
+
+// Stub for clGetCommandBufferInfoKHR that returns command queues array
+static cl_int clGetCommandBufferInfoKHR_testCommandBufferKhrGetCommandQueues(
+    cl_command_buffer_khr command_buffer,
+    cl_command_buffer_info_khr param_name,
+    size_t param_value_size,
+    void *param_value,
+    size_t *param_value_size_ret,
+    int num_calls)
+{
+    (void) num_calls;
+    TEST_ASSERT_EQUAL_PTR(make_command_buffer_khr(0), command_buffer);
+    TEST_ASSERT_EQUAL_HEX(CL_COMMAND_BUFFER_QUEUES_KHR, param_name);
+    TEST_ASSERT(param_value == NULL || param_value_size >= 3 * sizeof(cl_command_queue));
+    if (param_value_size_ret != NULL)
+        *param_value_size_ret = 3 * sizeof(cl_command_queue);
+    if (param_value != NULL)
+    {
+        cl_command_queue *command_queues = (cl_command_queue *)param_value;
+        command_queues[0] = make_command_queue(0);
+        command_queues[1] = make_command_queue(1);
+        command_queues[2] = make_command_queue(2);
+    }
+    return CL_SUCCESS;
+}
+
+void testCommandBufferInfoKHRCommandQueues()
+{
+#if defined(cl_khr_command_buffer)
+    clGetCommandBufferInfoKHR_StubWithCallback(clGetCommandBufferInfoKHR_testCommandBufferKhrGetCommandQueues);
+    clRetainCommandQueue_ExpectAndReturn(make_command_queue(0), CL_SUCCESS);
+    clRetainCommandQueue_ExpectAndReturn(make_command_queue(1), CL_SUCCESS);
+    clRetainCommandQueue_ExpectAndReturn(make_command_queue(2), CL_SUCCESS);
+
+    VECTOR_CLASS<cl::CommandQueue> command_queues = commandBufferKhrPool[0].getInfo<CL_COMMAND_BUFFER_QUEUES_KHR>();
+    TEST_ASSERT_EQUAL(3, command_queues.size());
+    TEST_ASSERT_EQUAL_PTR(make_command_queue(0), command_queues[0]());
+    TEST_ASSERT_EQUAL_PTR(make_command_queue(1), command_queues[1]());
+    TEST_ASSERT_EQUAL_PTR(make_command_queue(2), command_queues[2]());
+
+    command_queues[0]() = nullptr;
```

Comment:
@bashbaug when I look at this obfuscated code I really think we lack this interface in the `cl::Wrapper`
```suggestion
    command_queues[0].set(nullptr);
```
@Lukasz-Krakowiak-Mobica This is just for exposition.

Reply:
I am trying to have consistence impl with rest of the the source. Do you really want me to change this ?

Reply:
I'm OK adding a `set()` function in the wrapper if we want one, but we could also just write:

```c++
command_queues[0] = nullptr;
```

Unless I'm missing something there's no reason to call `operator()`.... and actually it is almost certainly a bug to do so because it won't properly release reference counts on the object.

Reply:
Ah good point. Then perhaps `operator()` should be `const` to avoid such bugs.

Reply:
I think it isn't a bug. This is UT, and it helps only to prevent executing release() method: https://github.com/KhronosGroup/OpenCL-CLHPP/blob/main/include/CL/opencl.hpp#L1881 and doesn't change clReleaseCommandQueue_refcount.
But still if you will have any related changes I can apply them.

Reply:
This code looks super convoluted, has no comment and a few people does not understand it.
Why do you want to circumvent the normal reference counting behavior?
We love comments in the code.
Think that ChatGPT will use your tests as a reference implementation to be used everywhere. :-)

Reply:
Note, this is a pretty common pattern we use elsewhere in these unit tests even if it's not best practice or something we'd expect to see in real code.  An additional comment wouldn't hurt but I don't think we should refactor the tests to avoid this code, or at least not in this PR.

Example from a different test:

```c++
    // Prevent destructor from interfering with the test
    d0() = NULL;
```

Reply:
Sorry guys, I used similar impl as it was done for other UTs.
I applied some changes for this UT, it should be more clear now.